pkg/operation/hybridbotanist/controlplane.go

```diff
@@ -161,6 +204,32 @@ func (b *HybridBotanist) DeployKubeAPIServer() error {
 				"memory": "4000Mi",
 			},
 		}
+	} else {
+		maxNodes := b.Shoot.GetNodeCount()
+		// Get current kube-apiserver deployment
+		existingAPIServerDeployment, err := b.K8sSeedClient.GetDeployment(b.Shoot.SeedNamespace, common.KubeAPIServerDeploymentName)
+		if err == nil {
+			// As kube-apiserver HPA manages the number of replicas in big clusters
+			// maintain current number of replicas
+			// otherwise keep the value to default
+			if maxNodes >= common.EnableHPANodeCount && *existingAPIServerDeployment.Spec.Replicas > 0 {
```

Comment:
Just to be safe: please verify that `*existingAPIServerDeployment.Spec.Replicas` is not `nil`.